internal/host/common.go

```diff
@@ -106,16 +107,24 @@ func GetHostnameAndRoleByIP(ip string, hosts []*models.Host) (string, models.Hos
 		}
 		if strings.Count(ip, ":") == 0 {
 			for _, i := range inv.Interfaces {
-				for _, ipv4 := range i.IPV4Addresses {
-					if ip == ipv4 {
+				for _, cidr := range i.IPV4Addresses {
```

Comment:
i think you should use https://github.com/openshift/assisted-service/blob/master/internal/network/machine_network_cidr.go#L174 to parse it